src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -39,16 +41,22 @@
             return Collections.emptyList();
         }
 
-        try (InputStream stream = getUrlDownload(getURLForQuery(query)).asInputStream()) {
+        URL urlForQuery;
+        try {
+            urlForQuery = getURLForQuery(query);
+        } catch (URISyntaxException | MalformedURLException e) {
+            LoggerFactory.getLogger(this.getClass()).info("Search URL {} is malformed", query);
```

Comment:
Since you rethrow the exception anyway, I don't think you need to add a logger statement here. It's the job of the caller to decide how to handle (i.e. log / show to user) the exception.

Reply:
I switch to debug. Better more logging than too less.

Reply:
OK, found some article in the net that one should either log or throw the excpetion --> https://howtodoinjava.com/best-practices/java-exception-handling-best-practices/#6

Therefore, introdcuging `String.format` for the exception message.